parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -526,7 +528,35 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  public static DictionaryPage readDictionary(Configuration conf, ColumnChunkMetaData meta, FSDataInputStream fin) throws IOException {
```

Comment:
I think this should be an instance method, return compressed bytes, or require a decompressor passed in. This version leaks resources by not calling release on the codec factory, which is now attached to the life-cycle of the ParquetFileReader. I recommend reusing that one.